testing/ef-tests/src/macros/fork_choice.rs

```diff
@@ -0,0 +1,256 @@
+#[macro_export]
+macro_rules! test_fork_choice {
+    ($path:ident) => {
+        paste::paste! {
+            #[cfg(test)]
+            #[allow(non_snake_case)]
+            mod [<tests_ $path>] {
+                use std::fs;
+
+                use alloy_primitives::{hex, map::HashMap, B256, hex::FromHex};
+                use ream_consensus::{
+                    attestation::Attestation,
+                    attester_slashing::AttesterSlashing,
+                    checkpoint::Checkpoint,
+                    deneb::{
+                        beacon_block::{BeaconBlock, SignedBeaconBlock},
+                        beacon_state::BeaconState,
+                    },
+                    execution_engine::{mock_engine::MockExecutionEngine, rpc_types::get_blobs::{Blob, BlobsAndProofV1}}, polynomial_commitments::kzg_proof::KZGProof,
+                };
+                use ream_fork_choice::{
+                    handlers::{on_attestation, on_attester_slashing, on_block, on_tick},
+                    store::{get_forkchoice_store, Store},
+                };
+                use rstest::rstest;
+                use serde::Deserialize;
+                pub const MAX_BLOB_COMMITMENTS_PER_BLOCK: u64 = 4096;
+                use ssz_derive::{Decode, Encode};
+                use ssz_types::{
+                    typenum::{self, U1099511627776, U4096}, FixedVector, VariableList
+                };
+                use tree_hash::TreeHash;
+
+                use super::*;
+                use $crate::utils;
+
+                #[derive(Debug, Deserialize)]
+                pub struct Tick {
+                    pub tick: u64,
+                    #[serde(default)]
+                    pub valid: Option<bool>,
+                }
+
+                #[derive(Debug, Deserialize)]
+                pub struct ShouldOverrideForkchoiceUpdate {
+                    pub validator_is_connected: bool,
+                    pub result: bool,
+                }
+
+                #[derive(Debug, Deserialize)]
+                pub struct Head {
+                    pub slot: u64,
+                    pub root: B256,
+                }
+
+                #[derive(Debug, Deserialize)]
+                pub struct Checks {
+                    #[serde(default)]
+                    pub head: Option<Head>,
+                    #[serde(default)]
+                    pub time: Option<u64>,
+                    #[serde(default)]
+                    pub justified_checkpoint: Option<Checkpoint>,
+                    #[serde(default)]
+                    pub finalized_checkpoint: Option<Checkpoint>,
+                    #[serde(default)]
+                    pub proposer_boost_root: Option<B256>,
+                    #[serde(default)]
+                    pub get_proposer_head: Option<B256>,
+                    #[serde(default)]
+                    pub should_override_forkchoice_update: Option<ShouldOverrideForkchoiceUpdate>,
```

Comment:
```suggestion
                #[derive(Debug, Deserialize)]
                pub struct Checks {
                    pub head: Option<Head>,
                    pub time: Option<u64>,
                    pub justified_checkpoint: Option<Checkpoint>,
                    pub finalized_checkpoint: Option<Checkpoint>,
                    pub proposer_boost_root: Option<B256>,
                    pub get_proposer_head: Option<B256>,
                    pub should_override_forkchoice_update: Option<ShouldOverrideForkchoiceUpdate>,
```


do you need default?